libr/util/buf.c

```diff
@@ -426,18 +426,47 @@ static int r_buf_cpy(RBuffer *b, ut64 addr, ut8 *dst, const ut8 *src, int len, i
 }
 
 static int r_buf_fcpy_at (RBuffer *b, ut64 addr, ut8 *buf, const char *fmt, int n, int write) {
-	ut64 len, check_len;
+	ut64 len, check_len, file_sz, readbytes;
 	int i, j, k, tsize, bigendian, m = 1;
 	if (!b || b->empty) return 0;
+#if 0
 	if (b->fd != -1) {
 		eprintf ("r_buf_fcpy_at not supported yet for r_buf_new_file\n");
 		return 0;
 	}
+#endif
+	//remove this and use a global way to get size using fd or add a field for file size in rbuffer.
+	ut64 curr = lseek(b->fd, 0, SEEK_CUR);
+	file_sz = lseek(b->fd, 0, SEEK_END);
+	lseek(b->fd, curr, SEEK_SET);
+	//remove upto this part.
+
 	if (addr == R_BUF_CUR)
 		addr = b->cur;
 	else addr -= b->base;
-	if (addr == UT64_MAX || addr > b->length)
+	if (addr == UT64_MAX)
 		return -1;
+	ut8 *tmp_buf = NULL;
+	readbytes = b->length;
+	if (addr > b->length) {
```

Comment:
Here (https://github.com/radare/radare2/blob/master/libr/bin/format/elf/elf.c#L191-L192) it is sending `bin->ehdr.e_shnum`. So I think from name, it is number of section header.